### ISSUE TITLE
Remove custom media type for Git Signing API

### ISSUE THREAD

**Issue by gmlewis** [google/go-github#761]
GitHub Developer announcement:
https://developer.github.com/changes/2017-10-12-git-signing/

Theoretically, we can remove the custom media type `application/vnd.github.cryptographer-preview` but I have sent in a request to the GitHub Support team to clarify if removing this will cause problems for GitHub Enterprise customers. I will update this issue when I hear back.

**Comment by sahildua2305** [comment]
@gmlewis please feel free to assign this issue to me so that I get notified when we hear back.

**Comment by gmlewis** [comment]
Thank you, @sahildua2305!

**Comment by kshitij10496** [comment]
@gmlewis Have we heard back from the GitHub Support about this?

**Comment by gmlewis** [comment]
**TL;DR: One more week and we'll move forward on this issue if there are no compliants.**

Wow, sorry that I forgot to report back.

I did receive a response right away that the technical team would attempt to be very clear in all future announcements as to how GitHub Enterprise customers are affected by each custom media type removal, but I didn't get specific details about this one in particular.

I think I have a better understanding now, though, about GitHub Enterprise customers.
(If any Enterprise customers are listening in, please correct any misunderstandings I may have.)

It appears to me that GitHub Enterprise customers can "pin" their release to any version that is convenient for them to continue using. In other words, they are not forced to update their code to support the top-of-trunk (latest) API version. I get the feeling that this is because they are running their own private copy of the GitHub API and server on their own hardware (but I could be totally wrong here and would love to be educated).

So... if all of the above is correct, there is no way we can prevent breaking GitHub Enterprise customers who chose to remain on older versions of the GitHub API.

Now, if this is true, I think we should handle custom media-type removals in the following way... we make the issue (as we have always done), then have a cooling-off period where any GitHub Enterprise customer who cares about that custom media-type can request that we hold off removing it until they are ready for the change.

Note, however, that once #376 is resolved, this will totally be a non-issue because users of this package will be able to pin any tagged version they wish.

Let's therefore let this sit for one more week in case any GitHub Enterprise customer desired for us to leave the custom media-type. If we haven't heard from anyone in one week, let's proceed with this issue and remove it.

**Comment by palash25** [comment]
@gmlewis any updates on this? Is this issue still blocked because of versioning issue?

**Comment by gmlewis** [comment]
Thanks for the ping, @palash25.
Since #376 is resolved, we can move forward with this issue and tag the repo appropriately.
